uaclient/entitlements/fips.py

```diff
@@ -301,49 +301,53 @@ def packages(self) -> List[str]:
         packages = super().packages
         return self._replace_metapackage_on_cloud_instance(packages)
 
+    def _notice_remove(self, label: str, description: str):
+        try:
+            self.cfg.notice_file.remove(label, description)
+        except exceptions.NonRootUserError:
+            pass
+
+    def _notice_add(self, label: str, description: str):
+        try:
+            self.cfg.notice_file.add(label, description)
+        except exceptions.NonRootUserError:
+            pass
+
     def application_status(
         self,
     ) -> Tuple[ApplicationStatus, Optional[messages.NamedMessage]]:
         super_status, super_msg = super().application_status()
 
         if system.is_container() and not system.should_reboot():
-            self.cfg.notice_file.remove(
-                "", messages.FIPS_SYSTEM_REBOOT_REQUIRED.msg
-            )
+            self._notice_remove("", messages.FIPS_SYSTEM_REBOOT_REQUIRED.msg)
```

Comment:
Let's add a test here that shows that a non-root call where we should delete the notices, doesn't fail the `application_status` call